rust-client/src/lib.rs

```diff
@@ -297,6 +305,145 @@ pub struct EvidenceBlob {
     value: Vec<u8>,
 }
 
+/// Enumerates the four possible states that the service can be in.
+#[derive(Debug, PartialEq, serde::Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum ServiceState {
+    Down,
+    Initializing,
+    Ready,
+    Terminating,
+}
+
+/// This object models the state and capabilities of the verification API in the Veraison service.
+///
+/// An instance of this struct is returned from [`Discovery::get_verification_api()`].
+#[derive(serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct VerificationApi {
```

Comment:
I think `VerificationClient` (and `DiscoveryClient`) would sound more idiomatic, but this is just a personal preference.

Reply:
I think that "client" would be wrong for `VerificationApi`, because it's not an active object, only a description. You can't actually use it to make API calls, and I think thet `Client` would wrongly imply that you could. The `Discovery` object could legitimately be called `DiscoveryClient`, but this would be inconsistent with other active objects like `ChallengeResponse`, so we would want to change that everywhere. That would be a different PR, I think, if we want to do it.

Reply:
Ah, right, my bad, I thought it would be expanded to essentially become a client. Thanks for the info!